src/libsystemd/sd-bus/sd-bus.c

```diff
@@ -3605,3 +3605,20 @@ _public_ int sd_bus_is_monitor(sd_bus *bus) {
 
         return !!(bus->hello_flags & KDBUS_HELLO_MONITOR);
 }
+
+static void flush_close(sd_bus *bus) {
+        if (!bus)
+                return;
+
+        /* Flushes and closes the specified bus. We take a ref before,
+         * to ensure the flushing does not cause the bus to be
+         * unreferenced. */
+
+        sd_bus_flush_close_unref(sd_bus_ref(bus));
+}
+
+_public_ void sd_bus_default_flush_close(void) {
+        flush_close(default_starter_bus);
+        flush_close(default_user_bus);
+        flush_close(default_system_bus);
```

Comment:
Any particular reason to close them? Why not just `sd_bus_default_flush()`? Is there _every_ an advantage in closing the default bus? If so, I wonder why you used the default bus in that case.

With just using sd_bus_default_flush(), you could call it in multiple different paths without requiring to synchronize them.